pylint/extensions/_check_docs_utils.py

```diff
@@ -164,12 +162,20 @@ def possible_exc_types(node):
                     and isinstance(val, (astroid.Instance, nodes.ClassDef))
                     and utils.inherit_from_std_ex(val)
                 ):
-                    excs.append(val.name)
+                    excs.append(val)
 
-    try:
-        return {exc for exc in excs if not utils.node_ignores_exception(node, exc)}
-    except astroid.InferenceError:
-        return set()
+    final_exceptions = set()
+    for exc in excs:
+        try:
+            # can fail because it is trying to infer the handler of the exception
+            if utils.node_ignores_exception(node, exc.name):
+                continue
+        except astroid.InferenceError:
+            return set()
```

Comment:
I wonder if this should be `continue`. I know this is mimicking previous behaviour, but could it be better to continue instead of returning prematurely?

Reply:
Know clue what the side-effects are

Reply:
Would you be willing to try and see what happens with this change?

Reply:
Let's see what the test base says